liveblog.php

```diff
@@ -395,21 +456,44 @@ public static function is_viewing_liveblog_post() {
 	 * One of: 'enable', 'archive', false.
 	 */
 	public static function get_liveblog_state( $post_id = null ) {
-		if ( ! is_single() && ! is_admin() ) {
+		if ( ! is_single() && ! is_admin() && ! self::$is_rest_api_call) {
 			return false;
 		}
 		if ( empty( $post_id ) ) {
-			global $post;
-			if ( ! $post ){
-				return false;
+			if ( ! empty( self::$post_id ) ) {
+				$post_id = self::$post_id;
+			} else {
+				global $post;
+				if ( ! $post ){
+					return false;
+				}
+				$post_id = $post->ID;
 			}
-			$post_id = $post->ID;
 		}
 		$state = get_post_meta( $post_id, self::key, true );
 		// backwards compatibility with older values
 		if ( 1 == $state ) {
 			$state = 'enable';
 		}
+
+		//If Auto Archive is enabled,
+		if ( null !== self::$auto_archive_days ) {
+
+			//Lets grab todays day, convert it to a timestamp and look for any set auto archive date.
+			$today = date('Y-m-d H:i:s' );
+			$today_timestamp = strtotime($today);
+			$expiry = get_post_meta( $post_id, self::$auto_archive_expiry_key);
+
+			//if we have an expiry date lets compare them and if the
+			// expiry is less than today i.e. its in the past lets archive the liveblog.
+			if( $expiry ) {
+				if( (int)$expiry[0] < $today_timestamp ) {
+					self::set_liveblog_state( $post_id, 'archive' );
```

Comment:
Will this end up running on the front end?

Reply:
I believe it runs on, all page loads and the original feature PR had it that way. I guess ideally this sort of function would be a CRON job / scheduled task that would run and check once a day or so and run any housekeeping required, is there a preferred way to handle this?

Reply:
Possibly, for now I'm just concerned with whether this will cause front-end DB writes which would be bad bad bad.